examples/auth/payload/src/app/(app)/_utilities/getUser.ts

```diff
@@ -0,0 +1,16 @@
+'use server'
+import type { User } from '@/payload-types'
+
+import { getPayloadHMR } from '@payloadcms/next/utilities'
+import { unstable_cache } from 'next/cache'
+
+import config from '../../../payload.config'
+
+export const getUser = unstable_cache(
+  async (headers: Headers): Promise<User | null> => {
+    const payload = await getPayloadHMR({ config })
+    const { user } = await payload.auth({ headers })
+    return user
+  },
+  ['payload-user'],
+)
```

Comment:
In Next.js, you may replace `unstable_cache` (which is deprecated) with `"use cache";`: https://nextjs.org/docs/canary/app/api-reference/directives/use-cache